Assets/Scripts/RocketDestruction.cs

```diff
@@ -0,0 +1,48 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class RocketDestruction : MonoBehaviour
+{
+    bool isDead = false;
+
+    public AudioSource explosionAudio;
+
+    public Rigidbody2D rocketRigidBody2D;
+
+    public SpriteRenderer rocketSpriteRenderer;
```

Comment:
Are you sure you want all those as `public`? I think you should use `[SerializeField] private` instead. I could be wrong though.